src/options.ts

```diff
@@ -205,6 +219,42 @@ class SenderOptions {
     }
   }
 
+  /**
+   * Resolves the protocol version, if it is set to 'auto'. <br>
+   * If TCP transport is used, the protocol version will default to 1.
+   * In case of HTTP transport the /settings endpoint of the database is used to find the protocol versions
+   * supported by the server, and the highest will be selected.
+   * @param options SenderOptions instance needs resolving protocol version
+   */
+  static async resolveAuto(options: SenderOptions) {
+    parseProtocolVersion(options);
+    if (options.protocol_version !== PROTOCOL_VERSION_AUTO) {
+      return options;
+    }
+
+    const url = `${options.protocol}://${options.host}:${options.port}/settings`;
+    const settings: {
+      config: { LINE_PROTO_SUPPORT_VERSION: number[] };
+    } = await fetchJson(url);
```

Comment:
What if we have a self-signed certificate configured? Then this HTTP request will fail while the subsequent ILP requests would have succeeded.

Reply:
Are there any new/existing tests around that?